docs/tutorials/tpu-cluster-setup.md

```diff
@@ -0,0 +1,135 @@
+# Setting up a TPU Cluster
+
+This guide will walk you through the steps to set up and manage a TPU cluster using Google Cloud Platform (GCP) and Ray.
+
+## Prerequisites
+
+Before you begin, ensure you have:
+- Completed the [basic installation](installation.md)
+- Access to a Google Cloud project with TPU quota
+- [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) installed
+- [Ray](https://docs.ray.io/en/latest/installation.html) installed
+
+## TPU Cluster Architecture
+
+A typical TPU cluster consists of:
+- **Head Node**: A persistent GCP VM that coordinates the cluster
+- **Worker Nodes**: Autoscaling TPU VMs that execute the actual computation
+
+## Cluster Configuration
+
+The cluster configuration is defined in a [YAML template file](../../infra/marin-cluster-template.yaml). This template allows you to customize various aspects of your cluster:
+
+- Machine types for head and worker nodes
+- Number of CPUs and amount of RAM
+- Disk size and type
+- TPU types and versions
+- Minimum and maximum worker counts
+- Docker images and initialization commands
+- Network configuration
+- Environment variables and secrets
+
+For our clusters, each config is in a separate file in the `infra` directory. These files are automatically generated by the
+`infra/update-cluster-configs.py` script, which reads the `infra/marin-cluster-template.yaml` file.
+More information can be found in the [infra/README.md](../../infra/README.md).
+
+## Setting Up the Cluster
+
+1. Install Marin with TPU support:
+   ```bash
+   pip install -e ".[tpu]"
+   ```
+
+2. Set up your GCP environment:
+   ```bash
+   # Configure Docker for GCP artifact registry
+   gcloud auth configure-docker us-central2-docker.pkg.dev
+   gcloud auth configure-docker europe-west4-docker.pkg.dev
+   gcloud auth configure-docker us-west4-docker.pkg.dev
+   ```
+
+3. Start the cluster (replace `$CLUSTER` with your desired region, e.g., `us-central2`):
+   ```bash
+   ray up -y infra/marin-$CLUSTER.yaml
+   ```
+
+## Executing a Job on the cluster
+
+### Using Ray Submit
+
+1. Connect to the cluster using ray dashboard:
+   ```bash
+   ray dashboard infra/marin-$CLUSTER.yaml
+   ```
+2. In another terminal, submit a job using ray submit:
+   ```bash
+   ray job submit --working-dir . -- python experiments/tutorials/hello_world.py
+   ```
+
+### Using Ray Run
+
+1. Connect to the cluster using ray dashboard:
+   ```bash
+   ray dashboard infra/marin-$CLUSTER.yaml
+   ```
+2. We have made a thin wrapper on top of `ray submit` called [ray-run](../../marin/run/ray_run.py) which can be used to easily specify the pip requirements and environment variables.
+   ```bash
+   python marin/run/ray_run.py --env_vars WANDB_API_KEY ${WANDB_API_KEY}  --pip_deps jax==0.4.35,async-lru --  python experiments/tutorials/hello_world.py
```

Comment:
say this is specifying additional dependencies?

What variables should be passed in?  Ideally, you don't need to specify any new packages/environment variables, right?

Reply:
Yup, will add the correct explanation ,something like:

"You can specify additional dependencies and env variables apart from those set in the docker image using this script....."